dropshot/src/error.rs

```diff
@@ -107,9 +108,18 @@ pub struct HttpError {
  * deserialize an HTTP response corresponding to an error in order to access the
  * error code, message, etc.
  */
-#[derive(Debug, Deserialize, Serialize)]
+/*
+ * TODO: does this need to be pub if it's going to be expressed in the OpenAPI
+ * output?
+ */
+#[derive(Debug, Deserialize, Serialize, JsonSchema)]
+#[schemars(rename = "Error")]
+#[schemars(description = "Error information from a response.")]
 pub struct HttpErrorResponseBody {
     pub request_id: String,
+    // The combination of default and required removes "nullable" from the
+    // OpenAPI-flavored JSON Schema output.
+    #[schemars(default, required)]
```

Comment:
@david-crespo per your comments; while many types by default interpret `Option<T>` as both optional and `nullable` I chose to just have this be optional since it will never be `null` and I find the schema to be more comprehensible without it.

Reply:
So integrating this change will cause Nexus to stop making it `null` sometimes? Or is there a change also required in Nexus?

Reply:
I'll add a test to confirm. In general, I prefer to omit a field rather than having a null field, but I realize that isn't necessarily right or even a broadly held opinion.